snippet.rb

```diff
@@ -0,0 +1,42 @@
+def fetch_data_from_openweathermap(city_name, api_key)
```

Comment:
I see several flaws with this method:
- I feel like having such long method makes it less cohesive as it is responsible for handling everything that is related to getting weather data.
- In general this method is too long and most of the programming languages have guidelines that lines of code in the method shouldn't exceed 10 in most of the cases.
- This method's name doesn't describe what it does to full extent as it not only fetches data, but also decorates it and builds DTO